app/actions/handlers.py

```diff
@@ -100,103 +101,138 @@ async def action_pull_observations(integration, action_config: PullObservationsC
     if devices:
         logger.info(f"Devices pulled with success. Length: {len(devices)}")
 
-        # Get 1 day of data ONLY if no device state is set
-        start_time_limit = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days=1)
-
         for i, device in enumerate(devices):
-            logger.info(f'Processing device {device[0]} ({i + 1}/{len(devices)} for integration {integration.name})')
+            logger.info(f"Sending PubSub message to trigger 'pull_observations' for device {device[0]}.")
 
             device_id = device[0]
             device_name = device[1]
 
-            try:
-                device_state = await state_manager.get_state(
-                    str(integration.id),
-                    "pull_observations",
-                    device_id
-                )
-                if device_state:
-                    device_state = client.DeviceState.parse_obj(device_state)
-                    # Assign recorded_at_field_name value from config to current state
-                    device_state.recorded_at_field_name = action_config.recorded_at_field_name.value
-                else:
-                    device_state = client.DeviceState(
-                        recorded_at=start_time_limit,
-                        recorded_at_field_name=action_config.recorded_at_field_name.value
-                    )
-            except pydantic.ValidationError as e:
-                logger.info(f"Invalid device state for device {device_id}. Exception: {e}")
-                device_state = client.DeviceState(
-                    recorded_at=start_time_limit,
-                    recorded_at_field_name=action_config.recorded_at_field_name.value
-                )
-            except redis.exceptions.RedisError as e:
-                logger.info(f"Error while reading device state from cache. Device {device_id}. Exception: {e}")
-                device_state = client.DeviceState(
-                    recorded_at=start_time_limit,
-                    recorded_at_field_name=action_config.recorded_at_field_name.value
-                )
-
-            # Ensure we don't go back further than 24 hours
-            start_at = max(device_state.recorded_at, start_time_limit)
-
-            traccar_observations = await client.get_positions_since(
-                integration, start_at, device_id, device_state.recorded_at_field_name
+            config = {
+                "device_id": device_id,
+                "device_name": device_name,
+                "recorded_at_field_name": action_config.recorded_at_field_name.value
+            }
+            await publish_event(
```

Comment:
The implementation of  `publish_event` was recently changed and now it makes blocking io calls which won't play well with `asyncio`. I think we should review that (comments on [this thread](https://allenai.slack.com/archives/C039F6LM01W/p1724759858156739))